examples/fluids/qfunctions/newtonian.h

```diff
@@ -783,4 +785,58 @@ CEED_QFUNCTION(PressureOutflow_Jacobian_Prim)(void *ctx, CeedInt Q,
   return PressureOutflow_Jacobian(ctx, Q, in, out, StateFromY, StateFromY_fwd);
 }
 
+// *****************************************************************************
+// This QFunction implements the Harten Lax VanLeer (HLL) approximate Riemann
+// solver. Taking in two states.
+// *****************************************************************************
+CEED_QFUNCTION_HELPER StateConservative Harten_Lax_VanLeer_Flux(NewtonianIdealGasContext gas,
+    State left, State right, const CeedScalar normal[3]) {
+  // *INDENT-OFF*
+  // Inputs will be uleft and uright states
+
+  // Output will be the flux
+
+  // *INDENT-ON*
+  // Context
+
+  StateConservative flux_left = FluxInviscidDotNormal(gas, left, normal);
+  StateConservative flux_right = FluxInviscidDotNormal(gas, right, normal);
+
+  // Compute HLL flux
+  StateConservative Flux_HLL[3];
+  for (CeedInt i=0; i<3; i++) {
+  // compute speed
+
+  // change in states
+  CeedScalar deltas = right[i] - left[i];
+  CeedScalar s_left =  Dot3(left.Y.velocity, normal) - SoundSpeed(gas, left.Y.temperature);
+  CeedScalar s_right = Dot3(right.Y.velocity, normal) + SoundSpeed(gas, right.Y.temperature);
+  CeedScalar deltaF;
+  CeedScalar jumpspeed;
+  CeedScalar minspeed, maxspeed;
+  CeedScalar deltas1, deltas2;
+  if (fabs(deltas) <= 0.)
+    Flux_HLL[i] = flux_right;
+  else {
+    deltaF = flux_right - flux_left;
+    jumpspeed = deltaF/deltas;
+    minspeed = min(s_left,jumpspeed); // add the correct sleft
+    maxspeed = max(s_right,jumpspeed); // add the correct sright
+    if (maxspeed > minspeed) {
+      deltas1 = (maxspeed*deltas - deltaF)/(maxspeed - minspeed);
+      deltas2 = (deltaF - minspeed*deltas)/(maxspeed - minspeed);
+      Flux_HLL[i] = 0.5 *(flux_left + flux_right - fabs(minspeed)*deltas1 - fabs(maxspeed)*deltas2);
```

Comment:
There is no average in the HLL equation. It's just the one formula using `s_left`, `s_right`, `left`, `right`, `flux_left`, `flux_right`.